lib/bots/versionbot.ts

```diff
@@ -654,6 +713,124 @@ export class VersionBot extends GithubBot.GithubBot {
+            // Then the status checks have failed and we throw an error.
+            _.each(statuses.statuses, (status) => {
+                if (_.includes(contexts, status.context)) {
+                    if (status.state === 'success') {
+                        foundStatuses.push(true);
+                    } else {
+                        foundStatuses.push(false);
+                    }
+                }
+            });
+            if ((foundStatuses.length !== contexts.length) ||
+                _.includes(foundStatuses, false)) {
+                return false;
+            }
+
+            return true;
+        });
+    }
+
+    // Has VersionBot already made commits to the branch.
+    private versionBotCommits(prInfo: GithubBotApiTypes.PullRequest) {
+        const githubApi = this.githubApi;
+        const owner = prInfo.head.repo.owner.login;
+        const repo = prInfo.head.repo.name;
+
+        // Get the list of commits for the PR, then get the very last commit SHA.
+        return this.gitCall(githubApi.repos.getCommit, {
+            owner,
+            repo,
+            sha: prInfo.head.sha
+        }).then((headCommit: GithubBotApiTypes.Commit) => {
+            const commit = headCommit.commit;
+            const files = headCommit.files;
+
+            if ((commit.committer.name === process.env.VERSIONBOT_NAME) &&
+            _.find(files, (file: GithubBotApiTypes.CommitFile) => {
+                return file.filename === 'CHANGELOG.md';
+            })) {
+                return commit.message;
+            }
+
+            return null;
+        });
+    }
+
+    // Actually carry out a merge.
+    private finaliseMerge = (prInfo: GithubBotApiTypes.PullRequest) => {
+        // We will go ahead and perform a merge if we see VersionBot has:
+        // 1. All of the status checks have passed on the repo
+        // 2. VersionBot has committed something with 'CHANGELOG.md' in it
+        const owner = prInfo.head.repo.owner.login;
+        const repo = prInfo.head.repo.name;
+
+        return this.checkStatuses(prInfo).then((statusesPassed) => {
+            if (statusesPassed) {
+                // Get the list of commits for the PR, then get the very last commit SHA.
+                return this.versionBotCommits(prInfo).then((commitMessage: string | null) => {
+                    if (commitMessage) {
+                        // We go ahead and merge.
+                        return this.mergeToMaster({
+                            commitVersion: commitMessage,
+                            owner,
+                            prNumber: prInfo.number,
+                            repoName: repo
+                        }).then(() => {
+                            // No tags here, just mention it in Flowdock so its searchable.
+                            // It's not an error so doesn't need logging.
+                            const flowdockMessage = {
```

Comment:
Super-small nitpick, but I guess there's no need to construct the `flowdockMessage` if `process.env.VERSIONBOT_FLOWDOCK_ADAPTER` isn't set?